cockatrice/src/client/ui/widgets/visual_deck_editor/visual_deck_editor_widget.cpp

```diff
@@ -0,0 +1,347 @@
+
+    groupByComboBox = new QComboBox();
+    QStringList groupProperties;
+    groupProperties << "maintype"
+                    << "colors"
+                    << "cmc"
+                    << "name";
+    groupByComboBox->addItems(groupProperties);
+    groupByComboBox->setMinimumWidth(300);
+    connect(groupByComboBox, QOverload<const QString &>::of(&QComboBox::currentTextChanged), this,
+            &VisualDeckEditorWidget::actChangeActiveGroupCriteria);
+    actChangeActiveGroupCriteria();
+
+    sortCriteriaButton = new SettingsButtonWidget(this);
+
+    sortLabel = new QLabel(sortCriteriaButton);
+
+    QStringList sortProperties = {"colors", "cmc", "name", "maintype"};
+    sortByListWidget = new QListWidget();
+    sortByListWidget->setSelectionMode(QAbstractItemView::SingleSelection);
+    sortByListWidget->setDragDropMode(QAbstractItemView::InternalMove);
+    sortByListWidget->setDefaultDropAction(Qt::MoveAction);
+
+    for (const QString &property : sortProperties) {
+        QListWidgetItem *item = new QListWidgetItem(property, sortByListWidget);
+        item->setFlags(item->flags() | Qt::ItemIsDragEnabled | Qt::ItemIsSelectable | Qt::ItemIsEnabled);
+    }
+
+    connect(sortByListWidget->model(), &QAbstractItemModel::rowsMoved, this,
+            &VisualDeckEditorWidget::actChangeActiveSortCriteria);
+    // connect(sortByListWidget, &QListWidget::itemChanged, this, &VisualDeckEditorWidget::actChangeActiveSortCriteria);
+    actChangeActiveSortCriteria();
+
+    sortByListWidget->setSizePolicy(QSizePolicy::Minimum, QSizePolicy::Minimum);
+
+    sortCriteriaButton->addSettingsWidget(sortLabel);
+    sortCriteriaButton->addSettingsWidget(sortByListWidget);
+
+    displayTypeButton = new QPushButton(this);
+    connect(displayTypeButton, &QPushButton::clicked, this, &VisualDeckEditorWidget::updateDisplayType);
+
+    groupAndSortLayout->addWidget(groupByComboBox);
+    groupAndSortLayout->addWidget(sortCriteriaButton);
+    groupAndSortLayout->addWidget(displayTypeButton);
+
+    scrollArea = new QScrollArea();
+    scrollArea->setWidgetResizable(true);
+    scrollArea->setMinimumSize(0, 0);
+
+    // Set scrollbar policies
+    scrollArea->setHorizontalScrollBarPolicy(Qt::ScrollBarAlwaysOff);
+    scrollArea->setVerticalScrollBarPolicy(Qt::ScrollBarAsNeeded);
+
+    zoneContainer = new QWidget(scrollArea);
+    zoneContainerLayout = new QVBoxLayout(zoneContainer);
+    zoneContainer->setLayout(zoneContainerLayout);
+    scrollArea->addScrollBarWidget(zoneContainer, Qt::AlignHCenter);
+    scrollArea->setWidget(zoneContainer);
+
+    updateZoneWidgets();
+
+    cardSizeWidget = new CardSizeWidget(this);
+
+    mainLayout->addWidget(groupAndSortContainer);
+    mainLayout->addWidget(scrollArea);
+    mainLayout->addWidget(cardSizeWidget);
+
+    retranslateUi();
+}
+
+void VisualDeckEditorWidget::retranslateUi()
+{
+    sortLabel->setText(tr("Click and drag to change\nthe sort order within the groups"));
```

Comment:
I recognize the line break, but is this something we want handled in text or by a wrap method?